src/sql/src/vars.rs

```diff
@@ -0,0 +1,1499 @@
+        *self.max_objects_per_schema.value()
+    }
+
+    /// Returns the value of the `max_secrets` configuration parameter.
+    pub fn max_secrets(&self) -> u32 {
+        *self.max_secrets.value()
+    }
+
+    /// Returns the value of the `max_roles` configuration parameter.
+    pub fn max_roles(&self) -> u32 {
+        *self.max_roles.value()
+    }
+
+    /// Returns the value of the `max_result_size` configuration parameter.
+    pub fn max_result_size(&self) -> u32 {
+        *self.max_result_size.value()
+    }
+
+    /// Returns the value of the `allowed_cluster_replica_sizes` configuration parameter.
+    pub fn allowed_cluster_replica_sizes(&self) -> Vec<String> {
+        self.allowed_cluster_replica_sizes
+            .value()
+            .into_iter()
+            .map(|s| s.as_str().into())
+            .collect()
+    }
+
+    /// Returns the `persist_blob_target_size` configuration parameter.
+    pub fn persist_blob_target_size(&self) -> usize {
+        *self.persist_blob_target_size.value()
+    }
+
+    /// Returns the `persist_compaction_minimum_timeout` configuration parameter.
+    pub fn persist_compaction_minimum_timeout(&self) -> Duration {
+        *self.persist_compaction_minimum_timeout.value()
+    }
+
+    /// Returns the `crdb_connect_timeout` configuration parameter.
+    pub fn crdb_connect_timeout(&self) -> Duration {
+        *self.crdb_connect_timeout.value()
+    }
+
+    /// Returns the `dataflow_max_inflight_bytes` configuration parameter.
+    pub fn dataflow_max_inflight_bytes(&self) -> usize {
+        *self.dataflow_max_inflight_bytes.value()
+    }
+
+    /// Returns the `metrics_retention` configuration parameter.
+    pub fn metrics_retention(&self) -> Duration {
+        *self.metrics_retention.value()
+    }
+
+    /// Returns the `mock_audit_event_timestamp` configuration parameter.
+    pub fn mock_audit_event_timestamp(&self) -> Option<mz_repr::Timestamp> {
+        *self.mock_audit_event_timestamp.value()
+    }
+}
+
+/// A `Var` represents a configuration parameter of an arbitrary type.
+pub trait Var: fmt::Debug {
+    /// Returns the name of the configuration parameter.
+    fn name(&self) -> &'static str;
+
+    /// Constructs a flattened string representation of the current value of the
+    /// configuration parameter.
+    ///
+    /// The resulting string is guaranteed to be parsable if provided to
+    /// `Value::parse` as a [`VarInput::Flat`].
+    fn value(&self) -> String;
+
+    /// Returns a short sentence describing the purpose of the configuration
+    /// parameter.
+    fn description(&self) -> &'static str;
+
+    /// Returns the name of the type of this variable.
+    fn type_name(&self) -> &'static str;
+
+    /// Indicates wither the [`Var`] is visible for the given [`User`].
+    ///
+    /// Variables marked as `internal` are only visible for the
+    /// [`crate::catalog::SYSTEM_USER`] user.
+    fn visible(&self, is_sytem_user: bool) -> bool;
```

Comment:
Just checking because this is a large diff, are there other changes to this file besides this method?

Reply:
No semantic changes. Made some things public that wen're before. The diff now should be much easier to read